dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1712,15 +1711,13 @@ public Entry registerCheckerWorkflow(@ApiParam(hidden = true) @Parameter(hidden
 
             // Determine gitUrl
             gitUrl = tool.getGitUrl();
-
-            // Determine source control, org, and repo
-            Pattern p = Pattern.compile("git@(\\S+):(\\S+)/(\\S+)\\.git");
-            Matcher m = p.matcher(tool.getGitUrl());
-            if (m.find()) {
+            final Optional<Map<String, String>>  stringStringGitUrlMap = SourceCodeRepoFactory
+                    .parseGitUrl(gitUrl);
+            if (!stringStringGitUrlMap.isEmpty()) {
                 SourceControlConverter converter = new SourceControlConverter();
-                sourceControl = converter.convertToEntityAttribute(m.group(1));
-                organization = m.group(2);
-                repository = m.group(3);
+                sourceControl = converter.convertToEntityAttribute(stringStringGitUrlMap.get().get(SourceCodeRepoFactory.GIT_URL_SOURCE_KEY));
```

Comment:
You could create an intermediate variable for `stringStringGitUrlMap.get()`

Reply:
done